foundryup/install

```diff
@@ -12,7 +12,15 @@ BIN_PATH="$FOUNDRY_BIN_DIR/foundryup"
 
 # Create the .foundry bin directory and foundryup binary if it doesn't exist.
 mkdir -p $FOUNDRY_BIN_DIR
-curl -# -L $BIN_URL -o $BIN_PATH
+readonly CURL_FLAGS="${CURL_FLAGS:- -vvv --max-time 120 --retry 3 --fail}"
+
+# we deliberately don't quote CURL_FLAGS
+# shellcheck disable=SC2086 
+curl ${CURL_FLAGS}  "${BIN_UR}" -o "${BIN_PATH}"
+if ! $? -ne 0; then
```

Comment:
this looks like magic lol

defer to @onbjerg for this, but makes sense to me

Reply:
#2216 is not caused by anything in this file, and we have a similar check in main foundryup

Reply:
Which evidently is not working. Your assumption is that bash executes the file top down: it does not do that

Reply:
I never assumed that, I'm just stating that there is a similar check in foundryup that you could take a look at, as this patch does not fix #2216 - #2216 is not caused by anything in this file (as stated). I would ask you to please keep a non-combative tone